vowpalwabbit/cb_explore_adf_greedy.cc

```diff
@@ -50,19 +50,21 @@ cb_explore_adf_greedy::cb_explore_adf_greedy(float epsilon, bool first_only)
 void cb_explore_adf_greedy::update_example_prediction(multi_ex& examples)
 {
   ACTION_SCORE::action_scores& preds = examples[0]->pred.a_s;
-
   uint32_t num_actions = static_cast<uint32_t>(preds.size());
 
+  auto& ep_fts = examples[0]->_reduction_features.template get<reduction_features>();
```

Comment:
Hi,  A couple of questions...

1) Wondering how persistence would work for epsilon.  What happens during prediction for a stack that has non-stationary exploration reduction?

2) Wondering if it makes sense to grab epsilon from example, set _epsilon and let everything work as is?

Thoughts?

Reply:
1. That should depend on how the non-stationary reduction manages / updates epsilon. If the epsilon is set in reduction_features for an example it will always use that epsilon. 
2. Yes that would work too. I chose to not do that to keep the internal state of `cb_explore_adf_greedy` separate but it should be the same

Reply:
1. That should depend on how the non-stationary reduction manages / updates epsilon. If the epsilon is set in reduction_features for an example it will always use that epsilon. 
2. Yes that would work too. I chose to not do that to keep the internal state of `cb_explore_adf_greedy` separate but it should be the same